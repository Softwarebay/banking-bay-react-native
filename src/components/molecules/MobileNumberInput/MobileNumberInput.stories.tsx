import { text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react-native';
import { ViewStory } from 'utils/storybook/ViewStory';

import { MobileNumberInput } from './MobileNumberInput';

storiesOf('Molecules/MobileNumberInput', module)
	.addDecorator((getStory) => <ViewStory>{getStory()}</ViewStory>)
	.add('Default', () => (
		<MobileNumberInput
			flagIcon="flag-usa"
			placeholder={text('placeholder', '+17 | xxxxxxxxxx')}
			testID="MobileNumberInput-test-id"
		/>
	));
