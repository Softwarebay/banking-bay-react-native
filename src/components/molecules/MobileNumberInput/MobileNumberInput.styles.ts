import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
	container: {
		height: 50,
		paddingHorizontal: 20,
		paddingVertical: 14,
	},
	inputField: {
		flex: 1,
		marginLeft: 6,
	},
});
