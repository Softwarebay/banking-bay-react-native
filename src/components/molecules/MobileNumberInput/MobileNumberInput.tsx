import { forwardRef } from 'react';
import { StyleProp, ViewStyle } from 'react-native';

import {
	Card,
	Icon,
	IconProps,
	InputField,
	InputFieldHandle,
} from 'components/atoms';

import { styles } from './MobileNumberInput.styles';

interface Props {
	flagIcon: IconProps['name'];
	placeholder: string;
	testID: string;
	style?: StyleProp<ViewStyle>;
}

export const MobileNumberInput = forwardRef<InputFieldHandle, Props>(
	({ flagIcon, placeholder, testID, style }, ref) => {
		return (
			<Card style={[styles.container, style]}>
				<Icon name={flagIcon} size={20} />
				<InputField
					ref={ref}
					keyboardType="phone-pad"
					placeholder={placeholder}
					style={styles.inputField}
					testID={testID}
				/>
			</Card>
		);
	}
);
