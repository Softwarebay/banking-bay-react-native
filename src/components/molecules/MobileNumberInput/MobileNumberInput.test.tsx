import { fireEvent, render, screen } from '@testing-library/react-native';
import testRenderer from 'react-test-renderer';

import { MobileNumberInput } from './MobileNumberInput';

import type { Props } from './MobileNumberInput';

describe('MobileNumberInput', () => {
	const props: Props = {
		flagIcon: 'flag-usa',
		placeholder: 'Mobile number placeholder',
		testID: 'mobile-number-input-test-id',
		style: {
			height: 50,
		},
	};

	it('should match snapshot', () => {
		const tree = testRenderer.create(<MobileNumberInput {...props} />).toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('should render input field and icon', () => {
		render(<MobileNumberInput {...props} />);

		const mobileNumberInput = screen.getByTestId('mobile-number-input-test-id');
		const icon = screen.getByTestId('flag-usa-icon-test-id');
		const inputField = screen.getByTestId(
			'input-field-mobile-number-input-test-id'
		);

		expect(mobileNumberInput.children).toHaveLength(2);
		expect(icon).not.toBeNull();
		expect(inputField).not.toBeNull();
	});

	it('should display placeholder', () => {
		render(<MobileNumberInput {...props} />);

		const inputField = screen.getByTestId(
			'input-field-mobile-number-input-test-id'
		);

		expect(inputField.props).toHaveProperty(
			'placeholder',
			'Mobile number placeholder'
		);
	});

	it('should show numberpad', () => {
		render(<MobileNumberInput {...props} />);

		const inputField = screen.getByTestId(
			'input-field-mobile-number-input-test-id'
		);

		expect(inputField.props).toHaveProperty('keyboardType', 'phone-pad');
	});

	it('should apply additional styles', () => {
		render(<MobileNumberInput {...props} />);

		const mobileNumberInput = screen.getByTestId('mobile-number-input-test-id');

		expect(mobileNumberInput).toHaveStyle({
			height: 50,
		});
	});

	it('should handle type events', () => {
		render(<MobileNumberInput {...props} />);

		const inputField = screen.getByTestId(
			'input-field-mobile-number-input-test-id'
		);

		fireEvent.changeText(inputField, '123456');

		expect(inputField.props).toHaveProperty('value', '123456');
	});
});
