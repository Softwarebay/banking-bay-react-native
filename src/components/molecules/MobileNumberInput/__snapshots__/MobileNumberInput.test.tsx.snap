// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileNumberInput should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
      },
      Object {
        "flexDirection": "row",
      },
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 10,
          "shadowColor": "#161966",
          "shadowOffset": Object {
            "height": 6,
            "width": 0,
          },
          "shadowOpacity": 0.08,
          "shadowRadius": 16,
        },
        Array [
          Object {
            "height": 55,
            "paddingHorizontal": 20,
            "paddingVertical": 14,
          },
          Object {
            "height": 50,
          },
        ],
      ],
    ]
  }
  testID="mobile-number-input-test-id"
>
  <SvgMock
    height={20}
    testID="flag-usa-icon-test-id"
    width={20}
  />
  <TextInput
    autoCorrect={false}
    keyboardType="phone-pad"
    onChangeText={[Function]}
    placeholder="Mobile number placeholder"
    style={
      Array [
        Object {
          "color": "#1B1D4D",
          "fontFamily": "MulishRegular",
          "fontSize": 15,
        },
        Object {
          "flex": 1,
          "marginLeft": 8,
        },
      ]
    }
    testID="input-field-mobile-number-input-test-id"
    value=""
  />
</View>
`;
